### ISSUE TITLE
Custom HTTP Error Handler

### ISSUE THREAD

**Issue by nlanson** [nlanson/Yomi#17]
Maybe implement a custom HTTP error handler??

See:
https://pusher.com/tutorials/error-handling-angular-part-1#limitations-of-try-catch

  - Custom error class
  - + Error service?

**Comment by nlanson** [comment]
Change:
Just make DatabaseService return observable HTTP responses and handle errors with subscribe method.

Eg.
database.service.ts
```javascript
public function getSomething(): Observable<HTTPResponse<CommonAPIResponse>> {
    let url = this.baseurl + '/get/something/';
    return this.http.get<CommonAPIResult>(url, {observe: 'response'});
}
```

example.component.ts
```javascript
function something() {
    this.db.getSomething().subscribe(
        data => {
               this.something = data.body;
               //Do something with something
        },
        err => {
               console.log(e.error.message);
               //Do something with error
        }
    );
    //Do other shit unrelated to the observable contents.
}
```

**Comment by nlanson** [comment]
Implemented in 357f176f21d4dff19e025c3305a869a85846cb92
